### PR TITLE
Migrate to Marist environment and update libraries

### PR DESCRIPTION
**Pull request by JirkaAichler**
* Migration to Marist
* Update to SpringFox 3.0 - some breaking changes fixed
* Update APIML enabler to 1.13 - use the new configuration
* Update other dependencies
* Fix locking
* Fix truststores

### CODE REVIEW

File: .circleci/river.env [hunk 1]
```diff
@@ -2,7 +2,7 @@ export ZOWE_CLI_PROFILE_NAME="river"
 export TEST_PROFILE="zos"
 export ZOS_HOST="river.zowe.org"
 export ZOS_ZOSMF_PORT="10443"
-export ZOS_SSH_PORT="2022"
```
Comment (pj892031) [inline .circleci/river.env:5]: Why this change of River's configuration? Is it possible to use it in the future (or it is better to delete)?

File: zowe-api-dev/src/commands/zosbuild.ts [hunk 2]
```diff
@@ -42,21 +41,10 @@ export default class ZosBuild extends Command {
             execSshCommandWithDefaultEnv(projectConfig.buildCommand, zosDir, env);
             for (const [zosFile, targetFile] of Object.entries(projectConfig.buildFiles)) {
                 this.log(`Downloading ${zosDir}/${zosFile} to ${targetFile}`);
-                if (f.flags.curl) {
```
Comment (pj892031) [inline zowe-api-dev/src/commands/zosbuild.ts:45]: Why curl was removed?

File: zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apidoc/BasePathTransformationFilter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@RequiredArgsConstructor
+public class BasePathTransformationFilter implements WebMvcSwaggerTransformationFilter {
+
+    private final String basePath;
+
+    @Override
+    public Swagger transform(SwaggerTransformationContext<HttpServletRequest> context) {
+        Swagger swagger = context.getSpecification();
+
+        String originalBasePath = swagger.getBasePath() != null ? swagger.getBasePath() : "";
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apidoc/BasePathTransformationFilter.java:37]:
```suggestion
        String originalBasePath = swagger.getBasePath();
        if (originalBasePath == null) originalBasePath = "";
```

File: zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apidoc/BasePathTransformationFilter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                        entry -> entry.getKey().replaceAll("^" + basePath, ""),
+                        Map.Entry::getValue));
+        swagger.setPaths(newPaths);
+
+        return swagger;
+    }
+
+    @Override
+    public boolean supports(@NonNull DocumentationType delimiter) {
+        return delimiter == DocumentationType.SWAGGER_2;
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apidoc/BasePathTransformationFilter.java:51]:
```suggestion
        return DocumentationType.SWAGGER_2.equals(delimiter);
```

File: zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apiml/RegisterToApiLayer.java [hunk 5]
```diff
@@ -48,12 +57,54 @@ public void onApplicationEvent(ContextRefreshedEvent event) {
...
+
+    public String getApiGatewayBaseUrl() {
+        if ((apiGatewayBaseUrl == null) && (apiMediationClient != null)) {
+            Application gateway = apiMediationClient.getEurekaClient().getApplication("gateway");
+            if ((gateway != null) && !gateway.getInstances().isEmpty()) {
+                apiGatewayBaseUrl = gateway.getInstances().get(0).getHomePageUrl();
+                log.info("Using API Gateway at {}", apiGatewayBaseUrl);
+            }
+        }
+        return apiGatewayBaseUrl;
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apiml/RegisterToApiLayer.java:86]:
```suggestion
        if (apiGatewayBaseUrl != null) return apiGatewayBaseUrl;
        
        if (apiMediationClient == null) return null;
        Application gateway = apiMediationClient.getEurekaClient().getApplication("gateway");
        if ((gateway == null) || gateway.getInstances().isEmpty()) return null;
        apiGatewayBaseUrl = gateway.getInstances().get(0).getHomePageUrl();
        log.info("Using API Gateway at {}", apiGatewayBaseUrl);
        
        return apiGatewayBaseUrl;
```

to be more readable

File: zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java [hunk 6]
```diff
@@ -9,21 +9,129 @@
...
+        EurekaClient eurekaClient = mock(EurekaClient.class);
+        doReturn(eurekaClient).when(apiMediationClient).getEurekaClient();
+        doReturn(application).when(eurekaClient).getApplication("gateway");
+        doReturn(Collections.singletonList(instanceInfo)).when(application).getInstances();
+
+        assertSame(homepage, registerToApiLayer.getApiGatewayBaseUrl());
+        assertSame(homepage, registerToApiLayer.getApiGatewayBaseUrl());
+
+        verify(apiMediationClient, times(1)).getEurekaClient();
+        //verify(application, times(1)).getInstances();
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java:107]: Fix or remove

File: zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java [hunk 7]
```diff
@@ -9,21 +9,129 @@
...
+        RegisterToApiLayer registerToApiLayer = getMock(true);
+        doCallRealMethod().when(registerToApiLayer).getApiGatewayBaseUrl();
+
+        Application application = mock(Application.class);
+        EurekaClient eurekaClient = mock(EurekaClient.class);
+        doReturn(eurekaClient).when(apiMediationClient).getEurekaClient();
+        doReturn(application).when(eurekaClient).getApplication("gateway");
+        doReturn(Collections.emptyList()).when(application).getInstances();
+
+        assertSame(null, registerToApiLayer.getApiGatewayBaseUrl());
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java:122]:
```suggestion
        assertNull(registerToApiLayer.getApiGatewayBaseUrl());
```

File: zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java [hunk 8]
```diff
@@ -9,21 +9,129 @@
...
+
+        EurekaClient eurekaClient = mock(EurekaClient.class);
+        doReturn(eurekaClient).when(apiMediationClient).getEurekaClient();
+        //doReturn(null).when(eurekaClient).getApplication("gateway");
 
-    public void emptyConfigShouldFail() {
-        ApiMediationServiceConfig config = new ApiMediationServiceConfig();
-        Ssl ssl = new Ssl();
-        new RegisterToApiLayer().register(config, ssl);
+        assertSame(null, registerToApiLayer.getApiGatewayBaseUrl());
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java:134]:
```suggestion
        assertNull(registerToApiLayer.getApiGatewayBaseUrl());
```

File: .circleci/river.env (hunk 1 above)
Comment (plavjanik) [reply .circleci/river.env:5]: I think that we should delete it. My understanding is that River is dead and cannot be resurrected.

**Comment by plavjanik** [comment]
Thank you. It is a lot of great work. Pavel's suggestions are worth accepting.

File: .circleci/river.env (hunk 1 above)
Comment (JirkaAichler) [reply .circleci/river.env:5]: You still can connect to USS using SSH on River. However, zos configuration is broken. I still have seen some ideas to restore River configuration from backup. But they were a little bit vague.

File: zowe-api-dev/src/commands/zosbuild.ts (hunk 2 above)
Comment (plavjanik) [reply zowe-api-dev/src/commands/zosbuild.ts:45]: It was only a temporary workaround for a problem on River.

File: zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java (hunk 6 above)
Comment (plavjanik) [inline zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java:107]:
```suggestion
```

File: zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java [hunk 9]
```diff
@@ -9,21 +9,129 @@
...
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.zowe.apiml.eurekaservice.client.ApiMediationClient;
 
-import com.ca.mfaas.eurekaservice.client.config.ApiMediationServiceConfig;
-import com.ca.mfaas.eurekaservice.client.config.Ssl;
+import java.util.Collections;
 
-import org.junit.Test;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
```
Comment (plavjanik) [inline zowe-rest-api-commons-spring/src/test/java/org/zowe/commons/apiml/RegisterToApiLayerTests.java:25]:
```suggestion
import static org.junit.jupiter.api.Assertions.assertNotSame;
import static org.junit.jupiter.api.Assertions.assertNull;
```

**Pull request merged**